src/app/(sidebar)/smart-contracts/contract-explorer/components/InvokeContractForm.tsx

```diff
@@ -277,8 +287,12 @@ export const InvokeContractForm = ({
       // fetch sequence number first
       await fetchSequenceNumber();
 
-      if (!sequenceNumberData) {
-        throw new Error("Failed to fetch sequence number. Please try again.");
+      if (!sequenceNumberData || sequenceNumberError) {
+        const errorMessage =
+          sequenceNumberError ||
+          "Failed to fetch sequence number. Please try again.";
+
+        throw errorMessage;
```

Comment:
@quietbits this was added. I noticed that a generic simulation error was displaying instead of a specific `account not funded` error after the reset.